### PR TITLE
work around IE XHR bug with '' URL Fixes #618

### LINKED ISSUE #618: fetch with '' url rejects wth DOMException SyntaxError on IE
This is a bug in the IE XHR, another XHR wrapper has the same problem https://github.com/sstephenson/prototype/issues/234

### PR DESCRIPTION
**Pull request by graingert**
work around IE XHR bug with '' URL Fixes #618

### CODE REVIEW

**Comment by graingert** [comment]
@mislav can you take a look at this please?

**Comment by abdourahimjallow** [comment]
What is this about

**Comment by abdourahimjallow** [comment]
Tell me more please

**Comment by graingert** [comment]
Have a look at the issue for more details:
https://github.com/github/fetch/issues/618

File: fetch.js [hunk 1]
```diff
@@ -486,7 +486,15 @@ export function fetch(input, init) {
       reject(new DOMException('Aborted', 'AbortError'))
     }
 
-    xhr.open(request.method, request.url, true)
+    function fixUrl(url) {
+      try {
+        return url === '' && self.location.href ? self.location.href : url
+      } catch(e) {
```
Comment (JakeChampion) [inline fetch.js:492]:
```suggestion
      } catch (e) {
```

**Pull request merged**